packages/frontend/src/components/connectors/ConnectorCreate.vue

```diff
@@ -0,0 +1,253 @@
+            </div>
+          </template>
+        </b-steps>
+      </div>
+    </ConnectorModalLayout>
+  </div>
+</template>
+
+<script>
+import { cloneDeep } from 'lodash';
+import ConnectorModalLayout from '../../layouts/ConnectorModalLayout.vue';
+import DiscordSocket from './sockets/DiscordSocket.vue';
+import { notifySuccess, notifyFailure } from '../../notification';
+
+const loadingTemplate = {
+  text: '',
+  icon: '',
+  color: '',
+  active: true,
+};
+
+const connectorBlueprint = {
+  name: '',
+  type: '',
+  socket: {},
+};
+
+export default {
+  name: 'ConnectorCreate',
+  components: {
+    ConnectorModalLayout,
+  },
+  data: () => ({
+    activeStep: 0,
+    canClose: true,
+    connectorSocket: undefined,
+    connectorDraft: cloneDeep(connectorBlueprint),
+    isDirty: false,
+    loading: { ...loadingTemplate },
+    showCreate: false,
+  }),
+  methods: {
+    onClose() {
+      this.showCreate = false;
+
+      setTimeout(() => {
+        this.isDirty = false;
+        this.activeStep = 0;
+        this.connectorDraft = cloneDeep(connectorBlueprint);
+        this.connectorSocket = undefined;
+        this.loading = cloneDeep(loadingTemplate);
+      }, 700);
+    },
+    onCreate() {
+      this.activeStep++;
+
+      this.canClose = false;
+      this.loading.active = true;
+      this.$store.dispatch('createConnector', this.connectorDraft)
+        .then(() => {
+          this.setResult(true);
+          notifySuccess(this.$t('components.connectorCreate.notifications.connectorCreated'));
+          setTimeout(this.onClose, 2000);
+        })
+        .catch((apiResponse) => {
+          this.setResult(false);
+          setTimeout(() => { this.activeStep--; this.loading = loadingTemplate; }, 2000);
+          if (apiResponse.code) return notifyFailure(apiResponse.error[0].message);
+          // Request failed locally - maybe no internet connection etc?
+          return notifyFailure(this.$t('general.notifications.requestFailedLocally'));
+        }).finally(() => {
+          this.canClose = true;
+        });
+    },
+    onSelectType(type) {
+      this.isDirty = false;
+      switch (type) {
+        case 'discord':
+          console.log('Discord Bot');
+          this.connectorSocket = DiscordSocket;
+          this.connectorDraft.type = 'discord';
+          this.activeStep++;
+          break;
+        default:
+          console.warn(`${type} not implemented yet!`);
+          break;
+      }
+    },
+    setResult(success) {
+      if (success) {
+        this.loading.text = this.$t('components.connectorCreate.doneSuccessDescription');
+        this.loading.active = false;
+        this.loading.icon = 'check-circle';
+        this.loading.color = 'is-success';
+      } else {
+        this.loading.text = this.$t('components.connectorCreate.doneFailedDescription');
+        this.loading.active = false;
+        this.loading.icon = 'times-circle';
+        this.loading.color = 'is-danger';
+      }
```

Comment:
```suggestion
    setResult(success) {
      this.loading.active = false;
      if (success) {
        this.loading.text = this.$t('components.connectorCreate.doneSuccessDescription');
        this.loading.icon = 'check-circle';
        this.loading.color = 'is-success';
      } else {
        this.loading.text = this.$t('components.connectorCreate.doneFailedDescription');
        this.loading.icon = 'times-circle';
        this.loading.color = 'is-danger';
      }
```
The duplicate declaration of `loading.active = false` can be circumvented

Reply:
Fixed, but moved the line below to the if statement as it triggers ui changes